### DISCUSSION TITLE
Fiel upload

### DISCUSSION THREAD

**Discussion by sekhar-hari** [oobabooga/text-generation-webui#4218] [Q&A]
Hi - I am not sure if this feature is available in the Text Generation Web UI where it can connect to a local repository (ex: file system, or confluence, JIRA, or something similar), so that files can be uploaded to do Q&A / AI Search on those files. I'm new to this UI, and did not see this option. If it is already available, please let me know where to find. This is a much needed feature to perform Q&A on private data by connecting the UI to a locally hosted LLaMA2 70B model (not OpenAI ChatGPT).

Many thanks for any information you can offer on this topic.

**Comment by TechScorpion** [comment]
> This is a much needed feature to perform Q&A on private data by connecting the UI to a locally hosted LLaMA2 70B model (not OpenAI ChatGPT).

I don't think it's possible currently with Text-Generation-Webui. Hopefully they'll add it.

Something similar has been talked about in these threads where they talk about things like PrivateGPT:
https://github.com/oobabooga/text-generation-webui/discussions/3038
https://github.com/oobabooga/text-generation-webui/discussions/2057

Though that's a different app and as far as I know it's slow because it doesn't have the option to use the GPU.
I think there's a similar one called LocalGPT.

But ideally it can be added to this GUI and use the GPU.

**Comment by sekhar-hari** [comment]
Ok. I see a lot of extensions here: https://github.com/oobabooga/text-generation-webui/blob/main/docs/Extensions.md. I was hoping there will be one extension that can support the file upload feature.

**Reply by TechScorpion** [reply]
In the existing extensions you could see if Superbooga can do something like you want if you can get that working.
In the description in that link it says "taking as input text files, URLs, or pasted text"
That's a bit more info on it here: https://github.com/oobabooga/text-generation-webui/issues/2824
but it didn't seem to do what the user wanted in the above link

**Comment by sekhar-hari** [comment]
Ok, I will take a look. Is this feature in the roadmap? Do you know a timeline when this will be available? For example, something similar to this: https://www.danswer.ai/

**Reply by TechScorpion** [reply]
> Is this feature in the roadmap? Do you know a timeline when this will be available?

Sorry I don't know. I don't know if there's a roadmap of features of oobabooga (or the extensions) listed anywhere.
Hopefully someone else can post with more info
